homeassistant/components/mqtt/__init__.py

```diff
@@ -476,10 +476,13 @@ async def async_setup_entry(hass, entry):
     if conf is None:
         conf = CONFIG_SCHEMA({DOMAIN: dict(entry.data)})[DOMAIN]
     elif any(key in conf for key in entry.data):
-        _LOGGER.warning(
+        tmp_data = dict(entry.data)
+        tmp_data.pop(CONF_PASSWORD, None)
+        tmp_data.pop(CONF_USERNAME, None)
```

Comment:
Instead of popping it, should we replace it with `********` ? And maybe only password ?

Reply:
Also wonder if we can actually specify which keys are going to be overridden, not just show all the data ?

Reply:
OK, improved.
It will now print all keys present in both the config entry, and in `configuration.yaml`, including those which are implicitly present because they have default values.

It could print only keys which actually differ?